### ISSUE TITLE
Spinner/ScrollTop: SEO use HREF on anchor

### ISSUE THREAD

**Issue by djmj** [primefaces/primefaces#10191]
### Description

I was checking our ecommerce shop with google PageSpeed insights and wondered why our SEO score had taken a hit reducing it from 100% to 92% and the problem was _p:spinner_ up and down anchors.

The problem is a missing _href_ attribute which google punishes from SEO perspective. The crawler does not know its not a real anchor.

![grafik](https://github.com/primefaces/primefaces/assets/5265550/98144c71-fcce-4f98-b556-b429bae5e2d6)

**Other components**

Maybe other components are also affected?

### Describe the solution you would like

**Ideas**

- add rel = “nofollow” and maybe google will ignore this anchor
- Or better dont use anchor tag at all

**Comment by melloware** [comment]
Agreed, this should probably use an HREF like our other similar components do.

**Comment by djmj** [comment]
Thanks for fixing it. For curiousity, why use _a_ tag instead of _button_? 

Is there a special reason. Not sure if it would break css though.

**Comment by melloware** [comment]
every link in all of PrimeFaces uses the `href="#"`    I am not seeing it change the URL and scroll to the top.

**Comment by melloware** [comment]
I just fixed it using https://www.freecodecamp.org/news/javascript-void-keyword-explained/

**Comment by djmj** [comment]
Hmm maybe i made a mistake overriding the renderer. I was also surprised. I will observe in PF 13.

**Comment by melloware** [comment]
yeah give this a try 

```java
writer.writeAttribute("href", "javascript:void(0)", null);
```

instead of

```java
writer.writeAttribute("href", "#", null);
```

**Comment by melloware** [comment]
No you are right something is still not right I forgot we can't use `javascript.void` because of CSP.
